In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from ydata_profiling import ProfileReport

%matplotlib inline
sns.set_style('whitegrid')
plt.rcParams['figure.dpi'] == 300

False

In [7]:
# reading the dataset
df = pd.read_csv('../data/raw/house_prices.csv')
df.sample(5)

area_type   availability  ... balcony   price
3178   Super built-up  Area  Ready To Move  ...     2.0   60.60
11288  Super built-up  Area  Ready To Move  ...     2.0   50.00
8523         Built-up  Area  Ready To Move  ...     1.0   38.25
3770   Super built-up  Area  Ready To Move  ...     2.0  150.00
13091  Super built-up  Area  Ready To Move  ...     1.0   41.00

[5 rows x 9 columns]

In [11]:
# checking the shape of the data
df.shape

(13320, 9)

There are 13320 rows and 9 columns

In [13]:
# checking the dataframe information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13320 entries, 0 to 13319
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   area_type     13320 non-null  object 
 1   availability  13320 non-null  object 
 2   location      13319 non-null  object 
 3   size          13304 non-null  object 
 4   society       7818 non-null   object 
 5   total_sqft    13320 non-null  object 
 6   bath          13247 non-null  float64
 7   balcony       12711 non-null  float64
 8   price         13320 non-null  float64
dtypes: float64(3), object(6)
memory usage: 936.7+ KB


There are 3 numerical columns and 6 non-numerical columns in the dataset.

In [15]:
# looking at the summary statistics
df.describe().T

count        mean         std  min   25%   50%    75%     max
bath     13247.0    2.692610    1.341458  1.0   2.0   2.0    3.0    40.0
balcony  12711.0    1.584376    0.817263  0.0   1.0   2.0    2.0     3.0
price    13320.0  112.565627  148.971674  8.0  50.0  72.0  120.0  3600.0

Bath: 
- Most properties have between 1 to 3 bathrooms, with a median of 2. 
- There is a property with an unusually high number of bathrooms (maximum of 40), suggesting a potential outlier.

Balcony:
- The majority of properties have 1 to 2 balconies, with a median of 2.
- The presence of properties with 0 balconies indicates that some units might not have a balcony.

Price:
- Prices vary widely, with a mean of approximately 112.57 and a large standard deviation of 148.97.
- The majority of properties have prices between 8 and 120, as indicated by the interquartile range (IQR).
- There is a substantial difference between the 75th percentile and the maximum price (3600), suggesting the presence of potential outliers.